sys/event/callback.c

```diff
@@ -18,6 +18,7 @@ void _event_callback_handler(event_t *event)
 
 void event_callback_init(event_callback_t *event_callback, void (callback)(void *), void *arg)
 {
+    event_callback->super.list_node.next = NULL;
```

Comment:
Maybe

```suggestion
    memset(event_callback, 0, sizeof(*event_callback));
```

? That would initialize all fields, even if the struct gets new members later on. And since both GCC and clang use a builtin for `memset()`, both compilers will detect zeroing out members being set later on as dead writes, which are optimized out. So there is not additional cost to `memset()` on modern compilers, if all fields are written to / zeroed out anyway.

(But I guess this is more style / personal taste, even though my argument sounds at least to my ears convincing :upside_down_face:)

Reply:
If you affirm they are optimized out why not, I don't mind either way

Reply:
https://godbolt.org/z/M3b6PMsMx shows a minimum example with `memset()` where `memset()` is fully optimized out.